runbotics/runbotics-orchestrator-ui/src/src-app/utils/findVariablesInActivities.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    BPMNElement,
+    CamundaInputParameter,
+    CamundaOutputParameter,
+    CamundaParameter,
+    IBpmnGateway,
+    IBpmnSubProcessBusinessObject,
+    ISequenceFlowBusinessObject,
+    getParameterValue
+} from '#src-app/views/process/ProcessBuildView/Modeler/helpers/elementParameters';
+
+export const findVariablesInGateway = (businessObject: IBpmnGateway, searchPhrase: string): boolean => {
+    const gatewayOutgoingFlows: ISequenceFlowBusinessObject[] = businessObject.outgoing as unknown as ISequenceFlowBusinessObject[];
+
+    const gatewayExpressions = gatewayOutgoingFlows
+        .map(flow => flow.conditionExpression.body.toLowerCase())
+        .filter(expression => expression.includes(searchPhrase));
+
+    return gatewayExpressions.length > 0;
+};
+
+export const findVariablesInLoop = (element: BPMNElement, searchPhrase: string): boolean => {
+    const loopVariable = (element.businessObject as IBpmnSubProcessBusinessObject).loopCharacteristics.loopCardinality.body;
+    return loopVariable.toLowerCase().includes(searchPhrase);
+};
+export const findVariablesInAction = (element: BPMNElement, searchPhrase: string): boolean => {
+    const extensionElements = element.businessObject.extensionElements;
+
+    if (!extensionElements) return false;
+
+    const inputValues = extensionElements.values[0].inputParameters;
+    const outputValues = extensionElements.values[0].outputParameters;
+
+    if (
+        (inputValues && inputValues.length > 0 && findInputVariablesInAction(inputValues, searchPhrase)) ||
+        (outputValues && outputValues.length > 0 && findOutputVariablesInAction(outputValues, searchPhrase))
```

Comment:
what do you think about breaking this condition into two smaller ones e.g. `inputVariablesExist` and `outputVariablesExist` and then returning the alernative instead of returning `true` or `false` separately